lib/semantics/symbol.cc

```diff
@@ -530,6 +531,33 @@ static void DumpUniqueName(std::ostream &os, const Scope &scope) {
   }
 }
 
+// Dump flag list for a symbol in Unparse
+static void DumpFlagsForUnparse(std::ostream &os, const Symbol::Flags &flags) {
+  static std::unordered_map<Symbol::Flag, const std::string> flagMap{
+      {Symbol::Flag::Implicit, "implicit"},
+      {Symbol::Flag::LocalityLocal, "local"},
+      {Symbol::Flag::LocalityLocalInit, "local_init"},
+      {Symbol::Flag::LocalityShared, "shared"}};
+  std::size_t seen{0};
+
+  flags.IterateOverMembers([&](Symbol::Flag flag) {
+    if (flags.test(flag)) {
```

Comment:
Isn't `flags.IterateOverMembers` only iterating over the flags that are set in `flags`? This would make this test redundant.